`rosa create cluster` with mode auto and sts creates a prompt

The simple command `rosa create cluster --cluster-name test123 --region us-east-1 --mode auto --sts` is prompting to create operator roles and the oidc-provider due to no `--yes` flag being passed down the these additional create commands.

```
❯ rosa version
1.2.8
```

```
❯ rosa create cluster --cluster-name test123 --region us-east-1 --mode auto --sts
I: Using arn:aws:iam::1234567890123:role/ManagedOpenShift-Installer-Role for the Installer role
I: Using arn:aws:iam::1234567890123:role/ManagedOpenShift-ControlPlane-Role for the ControlPlane role
I: Using arn:aws:iam::1234567890123:role/ManagedOpenShift-Worker-Role for the Worker role
I: Using arn:aws:iam::1234567890123:role/ManagedOpenShift-Support-Role for the Support role
I: Creating cluster 'test123'
I: To view a list of clusters and their status, run 'rosa list clusters'
I: Cluster 'test123' has been created.
I: Once the cluster is installed you will need to add an Identity Provider before you can login into the cluster. See 'rosa create idp --help' for more information.
Name:                       test123
ID:                         1vn9ouvgp4hkqo5d7bfk8j
External ID:
Control Plane:              Customer hosted
OpenShift Version:
Channel Group:              stable
DNS:                        test123.63qu.p1.openshiftapps.com
AWS Account:               1234567890123
API URL:
Console URL:
Region:                     us-east-1
Multi-AZ:                   false
Nodes:
 - Control plane:           3
 - Infra:                   2
 - Compute:                 2
Network:
 - Type:                    OVNKubernetes
 - Service CIDR:            172.30.0.0/16
 - Machine CIDR:            10.0.0.0/16
 - Pod CIDR:                10.128.0.0/14
 - Host Prefix:             /23
STS Role ARN:               arn:aws:iam::1234567890123:role/ManagedOpenShift-Installer-Role
Support Role ARN:           arn:aws:iam::1234567890123:role/ManagedOpenShift-Support-Role
Instance IAM Roles:
 - Control plane:           arn:aws:iam::1234567890123:role/ManagedOpenShift-ControlPlane-Role
 - Worker:                  arn:aws:iam::1234567890123:role/ManagedOpenShift-Worker-Role
Operator IAM Roles:
 - arn:aws:iam::1234567890123:role/test123-m4x2-openshift-machine-api-aws-cloud-credentials
 - arn:aws:iam::1234567890123:role/test123-m4x2-openshift-cloud-credential-operator-cloud-credentia
 - arn:aws:iam::1234567890123:role/test123-m4x2-openshift-image-registry-installer-cloud-credential
 - arn:aws:iam::1234567890123:role/test123-m4x2-openshift-ingress-operator-cloud-credentials
 - arn:aws:iam::1234567890123:role/test123-m4x2-openshift-cluster-csi-drivers-ebs-cloud-credentials
 - arn:aws:iam::1234567890123:role/test123-m4x2-openshift-cloud-network-config-controller-cloud-cre
State:                      waiting (Waiting for OIDC configuration)
Private:                    No
Created:                    Nov  1 2022 09:44:31 UTC
Details Page:               https://console.redhat.com/openshift/details/s/2GwNppY9iN0qdzC
OIDC Endpoint URL:          https://rh-oidc.s3.us-east-1.amazonaws.com/1vn9ouvgp4hkqo5d7bfk8j

I: Preparing to create operator roles.
I: Creating roles using 'arn:aws:iam::1234567890123:user/osdCcsAdmin'
? Create the 'test123-m4x2-openshift-ingress-operator-cloud-credentials' role? (Y/n)
? Create the 'test123-m4x2-openshift-cluster-csi-drivers-ebs-cloud-credentials' role? (Y/n)
? Create the 'test123-m4x2-openshift-cloud-network-config-controller-cloud-cre' role? (Y/n)
? Create the 'test123-m4x2-openshift-machine-api-aws-cloud-credentials' role? (Y/n)
? Create the 'test123-m4x2-openshift-cloud-credential-operator-cloud-credentia' role? (Y/n)
? Create the 'test123-m4x2-openshift-image-registry-installer-cloud-credential' role? (Y/n)
I: Preparing to create OIDC Provider.
^C
```

Since the flag is marked as persistent, we haven't ran into this until today on the osde2e side, when I tried removing account role creation which we were setting `--yes`. 

Adding the flag to create cluster allows for it to be passed down correctly. 

https://github.com/openshift/rosa/blob/f316828defa40391dbbdd2bf7a6981666dd91fae/cmd/create/cluster/cmd.go#L1798-L1802

```
2022/11/01 05:04:52 cluster.go:213: Pruning `--service-cidr` and ``
2022/11/01 05:04:52 cluster.go:213: Pruning `--pod-cidr` and ``
2022/11/01 05:04:52 cluster.go:219: [--cluster-name osde2e-qw6dg --region us-east-1 --channel-group candidate --version 4.11.9-candidate --expiration-time 2022-11-01T16:04:52Z  --replicas 2    --host-prefix 0 --mode auto --yes --sts --properties InstalledVersion:openshift-v4.11.9-candidate --properties MadeByOSDe2e:true --properties UpgradeVersion:-- --properties Status:provisioning --properties JobName: --properties JobID:-1 --properties OwnedBy:bpratt]
Error: unknown flag: --yes
Usage:
  cluster [flags]

Examples:
  # Create a cluster named "mycluster"
  rosa create cluster --cluster-name=mycluster

  # Create a cluster in the us-east-2 region
  rosa create cluster --cluster-name=mycluster --region=us-east-2

Flags:
  -c, --cluster-name string                   Name of the cluster. This will be used when generating a sub-domain for your cluster on openshiftapps.com.
      --sts                                   Use AWS Security Token Service (STS) instead of IAM credentials to deploy your cluster.
      --role-arn string                       The Amazon Resource Name of the role that OpenShift Cluster Manager will assume to create the cluster.
      --external-id string                    An optional unique identifier that might be required when you assume a role in another account.
      --support-role-arn string               The Amazon Resource Name of the role used by Red Hat SREs to enable access to the cluster account in order to provide support.
      --controlplane-iam-role string          The IAM role ARN that will be attached to control plane instances.
      --worker-iam-role string                The IAM role ARN that will be attached to worker instances.
      --operator-roles-prefix string          Prefix to use for all IAM roles used by the operators needed in the OpenShift installer. Leave empty to use an auto-generated one.
      --multi-az                              Deploy to multiple data centers.
      --region string                         Use a specific AWS region, overriding the AWS_REGION environment variable.
      --version string                        Version of OpenShift that will be used to install the cluster, for example "4.3.10"
      --channel-group string                  Channel group is the name of the group where this image belongs, for example "stable" or "fast". (default "stable")
      --etcd-encryption                       Add etcd encryption. By default etcd data is encrypted at rest. This option configures etcd encryption on top of existing storage encryption.
      --http-proxy string                     A proxy URL to use for creating HTTP connections outside the cluster. The URL scheme must be http.
      --https-proxy string                    A proxy URL to use for creating HTTPS connections outside the cluster.
      --no-proxy strings                      A comma-separated list of destination domain names, domains, IP addresses or other network CIDRs to exclude proxying.
      --additional-trust-bundle-file string   A file contains a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store.
      --enable-customer-managed-key           Enable to specify your KMS Key to encrypt EBS instance volumes. By default account’s default KMS key for that particular region is used.
      --kms-key-arn string                    The key ARN is the Amazon Resource Name (ARN) of a CMK. It is a unique, fully qualified identifier for the CMK. A key ARN includes the AWS account, Region, and the key ID.
      --private-link                          Provides private connectivity between VPCs, AWS services, and your on-premises networks, without exposing your traffic to the public internet.
      --subnet-ids strings                    The Subnet IDs to use when installing the cluster. Format should be a comma-separated list. Leave empty for installer provisioned subnet IDs.
      --availability-zones strings            The availability zones to use when installing a non-BYOVPC cluster. Format should be a comma-separated list. Leave empty for the installer to pick availability zones
      --compute-machine-type string           Instance type for the compute nodes. Determines the amount of memory and vCPU allocated to each compute node.
      --replicas int                          Number of worker nodes to provision. Single zone clusters need at least 2 nodes, multizone clusters need at least 3 nodes. (default 2)
      --enable-autoscaling                    Enable autoscaling of compute nodes.
      --min-replicas int                      Minimum number of compute nodes. (default 2)
      --max-replicas int                      Maximum number of compute nodes. (default 2)
      --machine-cidr ipNet                    Block of IP addresses used by OpenShift while installing the cluster, for example "10.0.0.0/16".
      --service-cidr ipNet                    Block of IP addresses for services, for example "172.30.0.0/16".
      --pod-cidr ipNet                        Block of IP addresses from which Pod IP addresses are allocated, for example "10.128.0.0/14".
      --host-prefix int                       Subnet prefix length to assign to each individual node. For example, if host prefix is set to "23", then each node is assigned a /23 subnet out of the given CIDR.
      --private                               Restrict master API endpoint and application routes to direct, private connectivity.
      --disable-scp-checks                    Indicates if cloud permission checks are disabled when attempting installation of the cluster.
      --disable-workload-monitoring           Enables you to monitor your own projects in isolation from Red Hat Site Reliability Engineer (SRE) platform metrics.
      --watch                                 Watch cluster installation logs.
      --dry-run                               Simulate creating the cluster.
      --permissions-boundary string           The ARN of the policy that is used to set the permissions boundary for the operator roles in STS clusters.
  -m, --mode string                           How to perform the operation. Valid options are:
                                              auto: Resource changes will be automatic applied using the current AWS account

                                              manual: Commands necessary to modify AWS resources will be output to be run manually
  -i, --interactive                           Enable interactive mode.
  -o, --output string                         Output format. Allowed formats are [json yaml]
  -h, --help                                  help for cluster

2022/11/01 05:04:52 cluster.go:227: Error creating cluster: unknown flag: --yes
```

--- a/cmd/create/cluster/cmd.go
+++ b/cmd/create/cluster/cmd.go
@@ -544,6 +544,7 @@ func init() {
 	aws.AddModeFlag(Cmd)
 	interactive.AddFlag(flags)
 	output.AddFlag(Cmd)
+	confirm.AddFlag(flags)
 }
 
 func networkTypeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {

